### PR TITLE
Us 187 resolve firebase bug

### PR DESCRIPTION
**Pull request by Foebus**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt [hunk 1]
```diff
@@ -59,46 +62,47 @@ class FirebaseGameLobbyRepository : IGameLobbyRepository {
...
+        setPlayerReady(gameId, uid, true, cb)
     }
 
-    override fun setPlayerReady(gameId: String, uid: String, ready: Boolean) {
-        fs.collection(GAME_PARTICIPATION_COLLECTION).whereEqualTo("playerID", uid).get().addOnSuccessListener { documents ->
-            for (doc in documents) {
-                fs.collection(GAME_PARTICIPATION_COLLECTION).document(doc.id).update("ready", ready)
-            }
-        }
+    private fun setHelper(gameId: String, cb: UnitCallback, modifier: participationModifier){
```
Comment (Zorvalt) [inline app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt]: The name is not very clear. When I read setHelper, I don't get what this function does. Maybe something like "updateUserParticipation" ?

File: app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt [hunk 2]
```diff
@@ -59,46 +62,47 @@ class FirebaseGameLobbyRepository : IGameLobbyRepository {
...
     }
 
-    override fun setPlayerFaction(gameId: String, uid: String, faction: Faction) {
-        fs.collection(GAME_PARTICIPATION_COLLECTION).whereEqualTo("playerID", uid).get().addOnSuccessListener { documents ->
-            for (doc in documents) {
-                fs.collection(GAME_PARTICIPATION_COLLECTION).document(doc.id).update("faction", faction)
-            }
-        }
-    }
+    override fun setPlayerReady(gameId: String, uid: String, ready: Boolean, cb: UnitCallback) { setHelper(gameId, cb){it.ready = ready} }
```
Comment (Zorvalt) [inline app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt]: It seems like uid is not used anymore, maybe it should be added to setHelper ?

File: app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt [hunk 3]
```diff
@@ -59,46 +62,47 @@ class FirebaseGameLobbyRepository : IGameLobbyRepository {
...
 
-    override fun setPlayerTag(gameId: String, uid: String, tag: String) {
-        fs.collection(GAME_PARTICIPATION_COLLECTION).whereEqualTo("playerID", uid).get().addOnSuccessListener { documents ->
-            for (doc in documents) {
-                fs.collection(GAME_PARTICIPATION_COLLECTION).document(doc.id).update("tag", tag)
-            }
-        }
-    }
+    override fun setPlayerFaction(gameId: String, uid: String, faction: Faction, cb: UnitCallback) { setHelper(gameId, cb){it.faction = faction} }
```
Comment (Zorvalt) [inline app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt]: It seems like uid is not used anymore, maybe it should be added to setHelper ?

File: app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt [hunk 4]
```diff
@@ -59,46 +62,47 @@ class FirebaseGameLobbyRepository : IGameLobbyRepository {
...
+
+    override fun setPlayerTag(gameId: String, uid: String, tag: String, cb: UnitCallback) {setHelper(gameId, cb){it.tag = tag}}
```
Comment (Zorvalt) [inline app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt]: It seems like uid is not used anymore, maybe it should be added to setHelper ?

File: app/src/main/java/ch/epfl/sdp/lobby/game/GameLobbyActivity.kt [hunk 5]
```diff
@@ -99,20 +102,24 @@ class GameLobbyActivity : AppCompatActivity(), SwipeRefreshLayout.OnRefreshListe
...
+        var newReadyState by Delegates.notNull<Boolean>()
         if(myFaction == Faction.PREY && myTag == null) {
             gameLobbyBinding.txtPlayerReady.text = getString(R.string.you_are_not_ready)
-            repository.setPlayerReady(gameID, userID, false)
+            newReadyState = false
         } else {
             gameLobbyBinding.txtPlayerReady.text = getString(R.string.you_are_ready)
-            repository.setPlayerReady(gameID, userID, true)
+            newReadyState = true
+        }
```
Comment (Zorvalt) [inline app/src/main/java/ch/epfl/sdp/lobby/game/GameLobbyActivity.kt]: Maybe using an if-expression would be clearer here than a delegate :
val newReadyState = if(...) {
        gameLobbyBinding.txtPlayerReady...
        false
    else {
        gameLobbyBinding.txtPlayerReady...
        true
    }

**Review by Zorvalt — CHANGES_REQUESTED** [review]

**Review by Zorvalt — APPROVED** [review]

**Review by pauldms — APPROVED** [review]

**Pull request merged**